pkg/vm/vm_test.go

```diff
@@ -2083,6 +2083,43 @@ func TestUNPACKBadNotArray(t *testing.T) {
 	runWithArgs(t, prog, nil, 1)
 }
 
+func TestPACKMAPDuplicateKeys(t *testing.T) {
+	prog := makeProgram(opcode.PACKMAP)
+	vm := load(prog)
+
+	keys := []string{"duplicateKey", "uniqueKey", "duplicateKey", "anotherUniqueKey"}
+	values := []string{"value1", "value2", "value3", "value4"}
+
+	for i := range keys {
+		vm.estack.PushVal(values[i])
+		vm.estack.PushVal(keys[i])
+	}
+
+	vm.estack.PushVal(len(keys))
+	runVM(t, vm)
+
+	require.Equal(t, 1, vm.estack.Len())
+	packedItem := vm.estack.Pop().Item()
+	assert.IsType(t, &stackitem.Map{}, packedItem)
```

Comment:
Just `assert.Equal(t, stackitem.MapT, packedItem.Type())`, don't use reflection if it's possible.

Reply:
And BTW, use `require` instead of `assert`, it's a critical error.